pkg/trait/classpath_test.go

```diff
@@ -0,0 +1,219 @@
+func TestConfigureClasspathTraitInWrongIntegrationPhaseDoesNotSucceed(t *testing.T) {
+	trait, environment := createNominalTest()
+	environment.Integration.Status.Phase = v1alpha1.IntegrationPhaseError
+
+	configured, err := trait.Configure(environment)
+	assert.Nil(t, err)
+	assert.False(t, configured)
+}
+
+func TestConfigureClasspathTraitInWrongIntegrationKitPhaseDoesNotSucceed(t *testing.T) {
+	trait, environment := createNominalTest()
+	environment.IntegrationKit.Status.Phase = v1alpha1.IntegrationKitPhaseWaitingForPlatform
+
+	configured, err := trait.Configure(environment)
+	assert.Nil(t, err)
+	assert.False(t, configured)
+}
+
+func TestConfigureClasspathDisabledTraitDoesNotSucceed(t *testing.T) {
+	trait, environment := createNominalTest()
+	trait.Enabled = new(bool)
+
+	configured, err := trait.Configure(environment)
+	assert.Nil(t, err)
+	assert.False(t, configured)
+}
+
+func TestConfigureClasspathTraitPlaftormIntegrationKitLazyInstantiation(t *testing.T) {
+	trait, environment := createNominalTest()
+	environment.IntegrationKit = nil
+	environment.Integration.Namespace = "kit-namespace"
+	environment.Integration.Status.Kit = "kit-name"
+
+	err := trait.Apply(environment)
+
+	assert.Nil(t, err)
+	assert.Equal(t, strset.New("/etc/camel/resources", "./resources"), environment.Classpath)
+}
+
+func TestConfigureClasspathTraitExternalIntegrationKitLazyInstantiation(t *testing.T) {
+	trait, environment := createTestWithKitType(v1alpha1.IntegrationKitTypeExternal)
+	environment.IntegrationKit = nil
+	environment.Integration.Namespace = "kit-namespace"
+	environment.Integration.Status.Kit = "kit-name"
+
+	err := trait.Apply(environment)
+
+	assert.Nil(t, err)
+	assert.Equal(t, strset.New("/etc/camel/resources", "./resources", "/deployments/dependencies/*"), environment.Classpath)
+}
+
+func TestConfigureClasspathTraitWithIntegrationKitStatusArtifact(t *testing.T) {
+	trait, environment := createNominalTest()
+	environment.IntegrationKit.Status.Artifacts = []v1alpha1.Artifact{{ID: "", Location: "", Target: "/dep/target"}}
+
+	err := trait.Apply(environment)
+
+	assert.Nil(t, err)
+	assert.NotNil(t, environment.Classpath)
+	assert.Equal(t, strset.New("/etc/camel/resources", "./resources", "/dep/target"), environment.Classpath)
+}
+
+func TestConfigureClasspathTraitWithDeploymentResource(t *testing.T) {
+	trait, environment := createNominalTest()
+
+	container := corev1.Container{
+		VolumeMounts: []corev1.VolumeMount{{MountPath: "/mount/path"}},
+	}
+
+	deployment := appsv1.Deployment{
+		Spec: appsv1.DeploymentSpec{
+			Template: corev1.PodTemplateSpec{
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{container},
+				},
+			},
+		},
+	}
+
+	environment.Resources = kubernetes.NewCollection(&deployment)
+
+	err := trait.Apply(environment)
+
+	assert.Nil(t, err)
+	assert.NotNil(t, environment.Classpath)
+	assert.Equal(t, strset.New("/etc/camel/resources", "./resources"), environment.Classpath)
+
+	assert.Len(t, container.Env, 0)
```

Comment:
I wonder if we shouldn't expect a length of 1 here ? It would then possibly be a bug where modification happens in a copy of the deployment because of Visit() pass value by copy.

Reply:
I think the classpath should be set by the `container` trait instead of the `classpath` one which should be in charge only to compute `Environment.Classpath`

Reply:
@aldettinger do you have bandwidth to wotk on it ?

Reply:
Hey @lburgazzoli, thanks for review.
I don't have that much of bandwidth and would prefer focusing on completing tests for other traits in order to close #255.
Would it make sense that I merge this PR and create a separate issue to move the classpath setup in the contrainer trait ?

Reply:
goign to merge and I will work on the container trait if you don't mind